pinhead-client/src/main/java/org/candlepin/insights/pinhead/client/X509ApiClientFactory.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.insights.pinhead.client;
+
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.client.jaxrs.internal.ClientConfiguration;
+import org.jboss.resteasy.spi.ResteasyProviderFactory;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+
+/**
+ * Extends the generated ApiClient class to allow for X509 Client Certificate authentication
+ */
+public class X509ApiClientFactory {
```

Comment:
Are we going to add this factory to the Spring context somehow? So we can `@Autowire` `org.candlepin.insights.pinhead.client.ApiClient` ?

Reply:
When this is done, we will also need to be able configure the client via env/config properties.

Reply:
I assume so but I was going to leave that to a later commit.

Reply:
I like how the inventory api is wrapped by an InventoryService. Perhaps we could include a stub PinheadService in this PR? Once this is done, we should have everything getting injected correctly.

Reply:
I'll include both the Factory in spring context and PinheadService in my next PR.